portfolio/src/main/webapp/script.js

```diff
@@ -28,3 +28,23 @@ function addRandomFact() {
   // Add it to the page.
   document.getElementById('random-facts-container').innerText = randomFact;
 }
+
+function displayMessages() {
+  fetch('/data').then(response => response.json()).then((messageArray) => {
+    
+    const messageList = document.getElementById('message-container');
+    messageList.innerHTML = '';
+    messageList.appendChild(createListElement('1) ' + messageArray[0]));
+    messageList.appendChild(createListElement('2) ' + messageArray[1]));
+    messageList.appendChild(createListElement('3) ' + messageArray[2]));
+  })
+  .catch((error) => {
+  console.error('Error:', error);
+  });
+}
+
+function createListElement(text) {
+  const liElement = document.createElement('li');
```

Comment:
Add all the <li> elements under a <ol> element will cause it to be numbered, so you won't need to add the number in the text.
https://www.w3schools.com/tags/tag_ol.asp